### PR TITLE
doc(c_sharp_sdk):add en_US/c_sharp_sdk markdown file

### PR DESCRIPTION
**Pull request by wuzufeng**

### CODE REVIEW

File: en_US/connect_to_deployments/c_sharp_sdk.md [hunk 1]
```diff
@@ -1 +1,159 @@
...
+    System.Threading.Thread.Sleep(1*1000);        
+    string msg = "messages: " + msg_count.ToString();        
+    client.Publish(topic, System.Text.Encoding.UTF8.GetBytes(msg));        
+    Console.WriteLine("Send `{0}` to topic `{1}`", msg, topic);       
+    msg_count++;    
+  } 
+}
+```
+
+### Subscribe to messages
```
Comment (Zhiw) [inline en_US/connect_to_deployments/c_sharp_sdk.md:80]: 这里应该是订阅主题吧

File: en_US/connect_to_deployments/c_sharp_sdk.md [hunk 2]
```diff
@@ -1 +1,159 @@
...
+        {
+            string payload = System.Text.Encoding.Default.GetString(e.Message);
+            Console.WriteLine("Received `{0}` from `{1}` topic", payload, e.Topic.ToString());
+        }
+
+        static void Main(string[] args)
+        {
+            string broker = "broker.emqx.io";
+            int port = 1883;
+            string topic = "/Csharp/mqtt";
```
Comment (zibuyu1995) [inline en_US/connect_to_deployments/c_sharp_sdk.md:148]: topic 不要以 / 开头

**Review by zibuyu1995 — CHANGES_REQUESTED** [review]

File: en_US/connect_to_deployments/c_sharp_sdk.md [hunk 3]
```diff
@@ -1 +1,159 @@
 # Connect to Deployments with C# SDK
+[C#](https://docs.microsoft.com/en-us/dotnet/csharp/) is an **object-oriented** programming language provided by Microsoft that runs on **.NET Framework**.
+
+This article mainly introduces how to use the **paho.mqtt.m2mqtt** client and implement connection, subscribe, messaging, and other functions between the client and MQTT broker, in the C# project.
```
Comment (chenx-work) [inline en_US/connect_to_deployments/c_sharp_sdk.md:4]: This article mainly introduces how to use the **paho.mqtt.m2mqtt** client library in a C# project to connect, subscribe, publish and receive messages from the client to the MQTT server.

File: en_US/connect_to_deployments/c_sharp_sdk.md [hunk 4]
```diff
@@ -1 +1,159 @@
...
+### .NET CLI install M2Mqtt 
+Use the following command in the root of the project to install M2Mqtt.
+```bash
+dotnet add package M2Mqtt --version 4.3.0
+```
+
+## The use of C# MQTT
+
+### Connect to the MQTT broker
+This article will use [the free public MQTT broker](https://www.emqx.io/mqtt/public-mqtt5-broker) provided by EMQ X. This service is based on [MQTT IoT cloud platform](https://cloud.emqx.io/) to create. The accessing information of the broker is as follows:
```
Comment (chenx-work) [inline en_US/connect_to_deployments/c_sharp_sdk.md:25]: This service is created base on MQTT IoT cloud platform

File: en_US/connect_to_deployments/c_sharp_sdk.md [hunk 5]
```diff
@@ -1 +1,159 @@
...
+- TCP Port: **1883**
+- Websocket Port: **8083**
+
+### Import the M2Mqtt
+```c#
+using uPLibrary.Networking.M2Mqtt;
+```
+
+### Set the parameter of MQTT Broker connection
+Set the address, port and topic of MQTT Broker connection. At the same time, we call the C# `Guid.NewGuid()` to randomly generated uid as the MQTT client id.
```
Comment (chenx-work) [inline en_US/connect_to_deployments/c_sharp_sdk.md:36]: generate

File: en_US/connect_to_deployments/c_sharp_sdk.md [hunk 6]
```diff
@@ -1 +1,159 @@
...
+```c#
+string broker = "broker.emqx.io";
+int port = 1883;
+string topic = "Csharp/mqtt";
+string clientId = Guid.NewGuid().ToString();
+```
+
+### Write the MQTT connect method
+Write static class method ConnectMQTT to create MQTT client connect to the specified broker, and we can determine whether the client is connected successfully according to client's  `IsConnected`. Finally, method will return the client.
```
Comment (chenx-work) [inline en_US/connect_to_deployments/c_sharp_sdk.md:45]: Write static class method ConnectMQTT to create an MQTT client and connect it to the specified broker. We can determine whether the client is connected successfully according to the client's property of `IsConnected`. In the end, the client is returned.

File: en_US/connect_to_deployments/c_sharp_sdk.md [hunk 7]
```diff
@@ -1 +1,159 @@
...
+  else    
+  {        
+    Console.WriteLine("Failed to connect");    
+  }    
+  return client; 
+}
+```
+
+### Publish messages
+We define a while loop. In this loop we will set the MQTT client `Publish` method to send messages to the specified topic every second.
```
Comment (chenx-work) [inline en_US/connect_to_deployments/c_sharp_sdk.md:64]: to publish messages

**Review by Zhiw — APPROVED** [review]

File: en_US/connect_to_deployments/c_sharp_sdk.md [hunk 8]
```diff
@@ -1 +1,159 @@
...
+### .NET CLI install M2Mqtt 
+Use the following command in the root of the project to install M2Mqtt.
+```bash
+dotnet add package M2Mqtt --version 4.3.0
+```
+
+## The use of C# MQTT
+
+### Connect to the MQTT broker
+This article will use [the free public MQTT broker](https://www.emqx.io/mqtt/public-mqtt5-broker) provided by EMQ X. This service is created base on [MQTT IoT cloud platform](https://cloud.emqx.io/) to create. The accessing information of the broker is as follows:
```
Comment (chenx-work) [inline en_US/connect_to_deployments/c_sharp_sdk.md:25]: 不好意思刚才打错了 应该是is created based on

**Review by chenx-work — APPROVED** [review]

**Review by zibuyu1995 — APPROVED** [review]

**Review by Adek06 — APPROVED** [review]

**Pull request merged**